src/features/search/components/SearchListHeader/SearchListHeader.tsx

```diff
@@ -0,0 +1,86 @@
+import { useRoute } from '@react-navigation/native'
+import React from 'react'
+import { ScrollViewProps } from 'react-native'
+import styled from 'styled-components/native'
+
+import { SearchGroupNameEnumv2 } from 'api/gen'
+import { UseRouteType } from 'features/navigation/RootNavigator/types'
+import { useSearchResults } from 'features/search/api/useSearchResults/useSearchResults'
+import { NumberOfResults } from 'features/search/components/NumberOfResults/NumberOfResults'
+import { AccessibilityRole } from 'libs/accessibilityRole/accessibilityRole'
+import { analytics } from 'libs/firebase/analytics'
+import { useGeolocation } from 'libs/geolocation'
+import { Banner } from 'ui/components/Banner'
+import { styledButton } from 'ui/components/buttons/styledButton'
+import { GenericBanner } from 'ui/components/ModuleBanner/GenericBanner'
+import { Touchable } from 'ui/components/touchable/Touchable'
+import { BicolorEverywhere as Everywhere } from 'ui/svg/icons/BicolorEverywhere'
+import { Error } from 'ui/svg/icons/Error'
+import { getSpacing, Spacer, Typo } from 'ui/theme'
+
+export interface SearchListHeaderProps extends ScrollViewProps {
+  nbHits: number
+}
+
+const logActivateGeolocfromSearchResults = () => {
+  analytics.logActivateGeolocfromSearchResults()
+}
+
+export const SearchListHeader: React.FC<SearchListHeaderProps> = ({ nbHits }) => {
+  const { position, showGeolocPermissionModal } = useGeolocation()
+  const { userData } = useSearchResults()
+  const { params } = useRoute<UseRouteType<'Search'>>()
+  const shouldDisplayUnavailableOfferMessage = userData && userData.length > 0
+  const unavailableOfferMessage = shouldDisplayUnavailableOfferMessage ? userData[0]?.message : ''
+
+  const onPress = () => {
+    logActivateGeolocfromSearchResults()
+    showGeolocPermissionModal()
+  }
```

Comment:
En vrai ici je ferai ça
```suggestion
export const SearchListHeader: React.FC<SearchListHeaderProps> = ({ nbHits }) => {
  const { position, showGeolocPermissionModal } = useGeolocation()
  const { userData } = useSearchResults()
  const { params } = useRoute<UseRouteType<'Search'>>()
  const shouldDisplayUnavailableOfferMessage = userData && userData.length > 0
  const unavailableOfferMessage = shouldDisplayUnavailableOfferMessage ? userData[0]?.message : ''

  const onPress = () => {
    analytics.logActivateGeolocfromSearchResults()
    showGeolocPermissionModal()
  }
```

Parce qu'on crée une fonction qui ne fait rien d'autre qu'en appeler une autre... Ca ne sert à rien.

PS: Puisque c'est difficilement visible avec Github, j'ai supprimé `const logActivateGeolocfromSearchResults` et j'ai remplacé son appel par `analytics.logActivateGeolocfromSearchResults()`